pkg/apis/migration/v1alpha1/migstorage_types.go

```diff
@@ -81,3 +84,73 @@ type BackupStorageConfig struct {
 func init() {
 	SchemeBuilder.Register(&MigStorage{}, &MigStorageList{})
 }
+
+// Determine if two BSLs are equal based on relevant fields in the Spec.
+// Returns `true` when equal.
+func (r *MigStorage) Equals(a, b *velero.BackupStorageLocation) bool {
+	return a.Spec.Provider == b.Spec.Provider &&
+		reflect.DeepEqual(a.Spec.Config, b.Spec.Config) &&
+		reflect.DeepEqual(
+			a.Spec.ObjectStorage,
+			b.Spec.ObjectStorage)
+}
+
+// Build a velero backup storage location.
+func (r *MigStorage) BuildBSL() *velero.BackupStorageLocation {
+	location := &velero.BackupStorageLocation{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       CorrelationLabels(r, r.Namespace, r.Name),
```

Comment:
So this will label our `BackupStorageLocation` with what is essentially an object reference to its owner, just in label format.

`'kind': 'namespace:name'`
`'MigStorage' : 'mig:sample-migstorage'`